sql-cli/src/sql_cli/sql_directory_parser.py

```diff
@@ -0,0 +1,129 @@
+from __future__ import annotations
+
+from pathlib import Path
+from typing import Iterable
+
+import frontmatter
+from sql_cli.utils import find_template_variables
+
+
+class SqlFile:
+    """
+    A SqlFile is equivalent to a transform step in the Astro SDK.
+
+    :param root_directory: The root directory path of the project.
+    :param path: The path to the sql file.
+    :param target_directory: The target directory path for the executable sql.
+    """
+
+    def __init__(
+        self, root_directory: Path, path: Path, target_directory: Path
+    ) -> None:
+        self.root_directory = root_directory
+        self.path = path
+        self.target_directory = target_directory
+
+        post = frontmatter.load(self.path)
+        self.content = post.content
+        self.metadata = post.metadata
+
+    def __eq__(self, __o: object) -> bool:
```

Comment:
why `__o` instead of just `other`?

Reply:
similar to what you already have in `__gt__`

Reply:
This was auto-generated for me when I had typed `def __eq` in my vscode 😅. I have updated it.